internal/cli/auth/login.go

```diff
@@ -52,18 +52,44 @@ const (
 	LogoutToLoginAccountMsg      = `run "atlas auth logout" first if you want to login with another Atlas account on the same Atlas CLI profile`
 )
 
-var errTimedOut = errors.New("authentication timed out")
-
 type LoginOpts struct {
 	cli.DefaultSetterOpts
-	AccessToken    string
-	RefreshToken   string
-	IsGov          bool
-	isCloudManager bool
-	NoBrowser      bool
-	SkipConfig     bool
-	config         LoginConfig
-	flow           Authenticator
+	AccessToken          string
+	RefreshToken         string
+	IsGov                bool
+	isCloudManager       bool
+	NoBrowser            bool
+	SkipConfig           bool
+	config               LoginConfig
+	flow                 Authenticator
+	regenerateCodePrompt userSurvey
+}
+
+func NewLoginOpts() *LoginOpts {
+	return &LoginOpts{
+		regenerateCodePrompt: &confirmPrompt{
+			message:         "Your one-time verification code is expired. Would you like to generate a new one?",
+			defaultResponse: true,
+		},
+	}
+}
+
+type userSurvey interface {
+	confirm() (response bool, err error)
+}
+
+type confirmPrompt struct {
+	message         string
+	defaultResponse bool
+}
+
+func (c *confirmPrompt) confirm() (response bool, err error) {
+	p := &survey.Confirm{
+		Message: c.message,
+		Default: c.defaultResponse,
+	}
+	err = survey.AskOne(p, &response)
```

Comment:
Please add a TODO for us to remember to update this for the telemetry version as per this PR https://github.com/mongodb/mongodb-atlas-cli/pull/1214/files

Reply:
can be a ticket as well IMO

Reply:
done, added a todo referencing CLOUDP-118532